packages/manager/apps/pci-databases-analytics/src/pages/services/[serviceId]/integrations/delete/DeleteIntegration.modal.tsx

```diff
@@ -44,41 +42,42 @@ const DeleteIntegration = ({
         variant: 'destructive',
         description: getCdbApiErrorMessage(err),
       });
-      if (onError) {
-        onError(err);
-      }
     },
     onSuccess: () => {
       toast.toast({
         title: t('deleteIntegrationToastSuccessTitle'),
         description: t('deleteIntegrationToastSuccessDescription', {
-          type: integration.type,
+          type: deletedIntegration.type,
         }),
       });
-      if (onSuccess) {
-        onSuccess(integration);
-      }
+      navigate('../');
     },
   });
 
+  useEffect(() => {
+    if (integrations && !deletedIntegration) navigate('../');
+  }, [integrations, deletedIntegration]);
+
   const handleDelete = () => {
     deleteIntegration({
       serviceId: service.id,
       projectId,
       engine: service.engine,
-      integrationId: integration.id,
+      integrationId: deletedIntegration.id,
     });
   };
 
   return (
-    <Dialog {...controller}>
+    <RouteModal isLoading={!integrations || !deletedIntegration}>
       <DialogContent>
         <DialogHeader>
           <DialogTitle data-testid="delete-integrations-modal">
             {t('deleteIntegrationTitle')}
           </DialogTitle>
           <DialogDescription>
-            {t('deleteIntegrationDescription', { type: integration.type })}
+            {t('deleteIntegrationDescription', {
+              type: deletedIntegration?.type,
```

Comment:
Could deletedIntegration be undefined here ?

Reply:
Yes, it can be on the first rendering